internal/pkg/conn/conn.go

```diff
@@ -95,14 +94,14 @@ func extractNatsTopic(topic string) string {
 }
 
 // handleMsg triggered when new message is received on MQTT broker
-func (b *broker) handleMsg(mc paho.Client, msg paho.Message) {
+func (b *broker) handleMsg(mc mqtt.Client, msg mqtt.Message) {
 	sm, err := senml.Decode(msg.Payload(), senml.JSON)
 	if err != nil {
 		b.logger.Warn(fmt.Sprintf("SenML decode failed: %s", err))
 		return
 	}
 
-	cmdType := cmdType(sm.Records[0].Name)
+	cmdType := sm.Records[0].Name
 	cmdStr := *sm.Records[0].StringValue
```

Comment:
why this pointer?

Reply:
because StringValue is pointer  to string so to fetch a string you have to dereference it
anyways, not my doing, it was here before me

Reply:
@manuio this is SenML library, should not be changed for now.